tests/staticfiles_tests/test_management.py

```diff
@@ -176,6 +176,40 @@ def test_handle_path_notimplemented(self):
         self.assertFileNotFound('cleared.txt')
 
 
+class TestInteractiveMessages(CollectionTestCase):
+    @staticmethod
+    def mock_input(stdout):
+        def _input(msg):
+            # Python2 reads bytes from the console output, use bytes for the StringIO
+            stdout.write(msg.encode('utf-8') if six.PY2 else msg)
+            return 'yes'
+        return _input
+
+    def test_warning_when_clearing_staticdir(self):
+        stdout = six.StringIO()
+
+        self.run_collectstatic()
+        with mock.patch('django.contrib.staticfiles.management.commands.collectstatic.input',
+                        side_effect=self.mock_input(stdout)):
+            call_command('collectstatic', interactive=True, clear=True, stdout=stdout)
+
+        output = force_text(stdout.getvalue())
+        self.assertNotIn('This will overwrite existing files!', output)
```

Comment:
I removed the NotIn assertions in my edits because they are brittle since a typo in the message means they would inadvertently pass. I suppose if error messages were class attributes that would make it more robust, but it seems unlikely to me that a regression could be introduced such that both messages are displayed.